packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1519,7 +1519,11 @@ abstract class FlutterCommand extends Command<void> {
   ApplicationPackageFactory? applicationPackages;
 
   /// Gets the parsed command-line option named [name] as a `bool`.
-  bool boolArg(String name) => argResults?[name] as bool? ?? false;
+  /// This has been deprecated, use [boolArg] instead.
+  bool boolArgDeprecated(String name) => argResults?[name] as bool? ?? false;
+
+  /// Gets the parsed command-line option named [name] as a `bool?`.
+  bool? boolArg(String name) => argResults?[name] as bool?;
```

Comment:
This function still raises the same ArgumentError as the deprecated one (when the "name" is not present on argResults) I suggest updating this function to handle that case. I am not sure if you can do it with a ternary operator in one line. In the proposal of the issue is an example of how to do it.

I also suggest adding 3 new tests under https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/args_test.dart

- argResult is null (boolArg should return null)
- "name" not present in argResult (boolArg? should return null)
- "name" present in argResult (boolArg return true/false depending on value)

Reply:
I will attempt to write a test for this next week.

Reply:
@amanv8060 Here is an example you can use to write your tests https://github.com/flutter/flutter/pull/103231